cores/esp8266/HashBuilder.h

```diff
@@ -0,0 +1,44 @@
+#ifndef __ESP8266_HASH_BUILDER__
+#define __ESP8266_HASH_BUILDER__
+
+#include <WString.h>
+#include <Stream.h>
+#include <memory>
+#include <vector>
+
+class HashBuilder{
+  public:
+    void begin(void);
+    void add(uint8_t * data, uint16_t len);
+    void calculate(void);
+
+    void addHexString(const char * data);
+    bool addStream(Stream & stream, const size_t maxLen);
+    String toString(void);
+
+    // deprecated
+    void getBytes(uint8_t * output );
+    void getChars(char * output);
```

Comment:
suggest expressing deprecation with `__attribute__((deprecated))` Why are these deprecated, by the way?